pkg/gatewayserver/gatewayserver.go

```diff
@@ -384,7 +384,7 @@ var (
 // This method should only be used for request contexts.
 func (gs *GatewayServer) FillGatewayContext(ctx context.Context, ids ttnpb.GatewayIdentifiers) (context.Context, ttnpb.GatewayIdentifiers, error) {
 	ctx = gs.FillContext(ctx)
-	if ids.IsZero() {
+	if ids.IsZero() || ids.EUI != nil && ids.EUI.IsZero() {
```

Comment:
I use parenthesis instinctively, even while making this change. In past PRs, @johanstokking has asked me to remove parentheses since it's `C` like so I removed it here to avoid comments.